src/chunk.c

```diff
@@ -1051,6 +1186,172 @@ chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
 	return oid_list;
 }
 
+/* Implementation of show_chunks that also take the caller name
+ * as argument so error messages refer to the function
+ * that the user actually called. This is necessary because
+ * show_chunk_impl is used by drop_chunks and export_chunks
+ */
+Datum
+ts_chunk_show_chunks(PG_FUNCTION_ARGS)
+{
+	FuncCallContext *funcctx;
+	char	   *caller_name;
+
+	Cache	   *hypertable_cache;
+	Hypertable *ht;
+	Dimension  *time_dim;
+	Oid			time_dim_type = InvalidOid;
+
+	/*
+	 * contains the list of hypertables which need to be considred. this is a
+	 * list containing a single hypertable if PG_ARGISNULL(0) is false.
+	 * Otherwise, it will have the list of all hypertables in the system
+	 */
+	List	   *hypertables = NIL;
+	ListCell   *lc;
+	int			ht_index = 0;
+	long		num_chunks = 0;
+	MemoryContext oldcontext;
+	ChunkScanCtx **ctxs;
+	Chunk	  **chunks;
+	Chunk	  **current;
+
+
+	Oid			table_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Datum		older_than_datum = PG_GETARG_DATUM(1);
+	Datum		newer_than_datum = PG_GETARG_DATUM(2);
+
+	/*
+	 * get_fn_expr_argtype defaults to UNKNOWNOID if argument is NULL but
+	 * making it InvalidOid makes the logic simpler later
+	 */
+	Oid			older_than_type = PG_ARGISNULL(1) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 1);
+	Oid			newer_than_type = PG_ARGISNULL(2) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 2);
+
+	/*
+	 * chunks_return_srf is called even when it is not the first call but only
+	 * after doing some computation first
+	 */
+	if (!SRF_IS_FIRSTCALL())
+		return chunks_return_srf(fcinfo);
+
+	funcctx = SRF_FIRSTCALL_INIT();
+
+	caller_name = PG_NARGS() <= 3 ? "show_chunks" : NameStr(*PG_GETARG_NAME(3));
+
+	if (PG_NARGS() > 3 && PG_ARGISNULL(3))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("When calling the internal function for show_chunks "
+						"caller_name cannot be null")
+				 ));
+
+	if (older_than_type != InvalidOid &&
+		newer_than_type != InvalidOid &&
+		older_than_type != newer_than_type)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("older_than_type and newer_than_type should have the same type")
+				 ));
+
+	/*
+	 * Cache outside the if block to make sure cached hypertable entry
+	 * returned will still be valid in foreach block below
+	 */
+	hypertable_cache = hypertable_cache_pin();
+	if (PG_ARGISNULL(0))
+	{
+		hypertables = hypertable_get_all();
+	}
+	else
+	{
+		ht = hypertable_cache_get_entry(hypertable_cache, table_relid);
+		if (!ht)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("table \"%s\" does not exist or is not a hypertable",
+							get_rel_name(table_relid))
+					 ));
+		hypertables = list_make1(ht);
+	}
+
+	oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+	ctxs = palloc(sizeof(ChunkScanCtx *) * list_length(hypertables));
```

Comment:
There are a lot of `..ctx` and `ctxs` variables. For clarity, I suggest a more accurate name, like `chunk_scan_ctxs` or similar.